samcli/commands/deploy/auth_utils.py

```diff
@@ -0,0 +1,134 @@
+                # Currently just checking for presence of Auth, not the details.
+                # https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-function-apifunctionauth.html
+                # https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-function-httpapifunctionauth.html
+                if event.get("Properties", {}).get("Auth", False):
+                    auth_resource_list.append((sam_function.name, True))
+                # Is there any auth defined on the referred http api or serverless api through the `id` construct?
+                elif _auth_id(sam_functions.resources, event.get("Properties", {}), identifier):
+                    auth_resource_list.append((sam_function.name, True))
+                else:
+                    auth_resource_list.append((sam_function.name, False))
+
+
+def _auth_id(resources_dict, event_properties, identifier):
+    """
+
+    Parameters
+    ----------
+    resources_dict: dict
+        Resolved resources defined in the SAM Template
+    event_properties: dict
+        Properties of given event supplied to a function resource
+    identifier: str
+        Id: `ApiId` or `RestApiId`
+
+    Returns
+    -------
+    bool
+        Returns if the given identifier under the event properties maps to a resource and has authorization enabled.
+
+    """
+    resource_name = event_properties.get(identifier, "")
+    api_resource = resources_dict.get(resource_name, {})
+    return any(
+        [
+            api_resource.get("Properties", {}).get("Auth", False),
+            _auth_definition_body_and_uri(
+                definition_body=api_resource.get("Properties", {}).get("DefinitionBody", {}),
+                definition_uri=api_resource.get("Properties", {}).get("DefinitionUri", None),
+            ),
+        ]
+    )
+
+
+def _auth_definition_body_and_uri(definition_body, definition_uri):
+    """
+
+    Parameters
+    ----------
+    definition_body: dict
+        inline definition body defined in the template
+    definition_uri: string
+        Either an s3 url or a local path to a definition uri
+
+    Returns
+    -------
+    bool
+        Is security defined on the swagger or not?
+
+
+    """
+
+    reader = SwaggerReader(definition_body=definition_body, definition_uri=definition_uri)
+    swagger = reader.read()
+    _auths = []
+    if not swagger:
+        swagger = {}
+    # NOTE(sriram-mv): Authorization and Authentication is indicated by the `security` scheme.
+    # https://swagger.io/docs/specification/authentication/
+    for _, verb in swagger.get("paths", {}).items():
+        for _property in verb.values():
+            _auths.append(bool(_property.get("security", False)))
+
+    _auths.append(bool(swagger.get("security", False)))
+
+    # This is not an exhaustive check, but to check if there is some form of security setup.
+    return any(_auths)
```

Comment:
just a suggestion, can we have list of check methods which all take same parameter and return true of false after validating a type of auth. It might be easy to extend by simply adding more auth methods?

Reply:
agreed with a series of handler methods. or just a series of filters.